From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eddie Zaneski <eddiezane@gmail.com>
Date: Tue, 27 Jul 2021 11:46:45 -0600
Subject: --EKS-PATCH-- Fix kubectl version unit test

Cherry-pick of upstream Kubernetes:
https://github.com/kubernetes/kubernetes/pull/103955, which is
included in Kubernetes 1.23.

Fixes configuration requirement that results in test
TestNewCmdVersionWithoutConfigFile failing for some developers.

Signed-off-by: Kirsten Schumy <ksschumy@amazon.com>
---
 .../k8s.io/kubectl/pkg/cmd/version/version_test.go   | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go b/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go
index 5b1fbf1f581..16b3827c9be 100644
--- a/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/version/version_test.go
@@ -23,11 +23,12 @@ import (
 	"github.com/spf13/cobra"
 
 	"k8s.io/cli-runtime/pkg/genericclioptions"
-	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
 )
 
-func TestNewCmdVersionWithoutConfigFile(t *testing.T) {
-	tf := cmdutil.NewFactory(&genericclioptions.ConfigFlags{})
+func TestNewCmdVersionClientVersion(t *testing.T) {
+	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	defer tf.Cleanup()
 	streams, _, buf, _ := genericclioptions.NewTestIOStreams()
 	o := NewOptions(streams)
 	if err := o.Complete(tf, &cobra.Command{}); err != nil {
@@ -36,10 +37,7 @@ func TestNewCmdVersionWithoutConfigFile(t *testing.T) {
 	if err := o.Validate(); err != nil {
 		t.Errorf("Unexpected error: %v", err)
 	}
-	// FIXME soltysh:
-	// since we have defaulting to localhost:8080 in staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#getDefaultServer
-	// we need to ignore the localhost:8080 server, when above gets removed this should be dropped too
-	if err := o.Run(); err != nil && !strings.Contains(err.Error(), "localhost:8080") {
+	if err := o.Run(); err != nil {
 		t.Errorf("Cannot execute version command: %v", err)
 	}
 	if !strings.Contains(buf.String(), "Client Version") {
